qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/common/data_handling/test/test_data_presenter.py

```diff
@@ -379,7 +379,57 @@ def _setup_bgsub_test(self):
         self.presenter.row_numbers["name1"] = 0
         self.presenter.row_numbers["name2"] = 1
         self.model.get_loaded_workspaces.return_value = {"name1": self.ws1, "name2": self.ws2}
-        self.model.estimate_background.return_value = self.ws2
+        self.model.estimate_background.return_value = self.ws2, True
+
+    def test_handle_table_cell_changed_invalid_bg_params_on_niter(self):
+        self.presenter.row_numbers = data_presenter.TwoWayRowDict()
```

Comment:
These 3 tests added seem to have a similar setup - could these be put in a helper function that perhaps takes an argument to set `self.view.read_bg_params_from_table.return_value`?